### ISSUE TITLE
Define haskey on model

### ISSUE THREAD

**Issue by odow** [jump-dev/JuMP.jl#1887]
As suggested: https://discourse.julialang.org/t/jump-19-0-constraint-by-name-fails-for-vectorized-constraints/21481/4?u=odow

It would be nice if this worked
```julia
using JuMP
m = Model()
@variable(m, x)
@constraint(m, mycons, x == 0)
haskey(m, :x)  # true
haskey(m, :y)  # false
haskey(m, :mycons)  # true
haskey(m, 1)  # error("Model look-up via keys is only supported with symbols")
```

**Comment by mlubin** [comment]
We should think about using the new [`getproperty`](https://docs.julialang.org/en/v1/base/base/#Base.getproperty) and `setproperty!` overloading instead of treating the model as a pseudo-dictionary. This could look like:
```julia
m = Model()
@variable(m, x)
@constraint(m, mycons, x == 1)
m.objects.x # returns x
m.objects.y # error: does not exist
m.objects.y = 10 # you can stuff anything into `m.objects`
haskey(m.objects, :mycons) # true
haskey(m.objects, :foo) # fase
```

**Comment by odow** [comment]
It seems like this was implemented in #2363 but not closed. I guess the decision whether to close this is if we want to implement a `model.objects` API, or keep the quasi-dictionary API that we currently have:
```Julia
julia> using JuMP

julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x)
x

julia> @constraint(model, c, x == 1)
c : x = 1.0

julia> model[:x]
x

julia> model[:y]
ERROR: KeyError: key :y not found
Stacktrace:
 [1] getindex(::Model, ::Symbol) at /Users/oscar/.julia/dev/JuMP/src/JuMP.jl:1035
 [2] top-level scope at REPL[7]:1

julia> model[:y] = 10
10

julia> haskey(model, :c)
true

julia> haskey(model, :foo)
false

julia> model[:c]
c : x = 1.0

julia> model[:y]
10
```

**Comment by odow** [comment]
I'm closing this as won't fix. We seem to be getting on fine without it. The model acts like a dictionary. We don't need to follow Pyomo's lead here.
